### DISCUSSION TITLE
Testing the combobox

### DISCUSSION THREAD

**Discussion by gabrielpenhazup** [adobe/react-spectrum#3775] [General]
Hi, I'm trying to do the combobox component tests. But I can't test the visibility of the list and list items. I used fake timer and act. Does anyone have any suggestions?

**Comment by reidbarber** [comment]
Take a look at the setup in our ComboBox tests:

https://github.com/adobe/react-spectrum/blob/8fa814d27247aa6350c3f8a1bdd91ad00a721e03/packages/%40react-spectrum/combobox/test/ComboBox.test.js#L245-L269

Particularly mocking screen height/width.

Also, this is related to https://github.com/adobe/react-spectrum/issues/3703

**Reply by gabrielpenhazup** [reply]
I did some tests, and found the problem. When I am checking the state of the list the useAsyncList is still in loading state. I'm trying to mock him, without success yet.
I tested the combobox with a 'defined' list and it worked

**Reply by snowystinger** [reply]
if you're dealing with promises in tests and are using fakeTimers, you need to allow for the promise queue to be processed, this can be done in two ways

```
await act(async () => *advance timers*);
// or
act(() => Promise.resolve());
```

**Reply by gabrielpenhazup** [reply]
Thanks for the tip! I opted for the first option and it worked, I believe I can continue with the tests. Thank you to all participants of this help!
